### PR TITLE
fix: system sessions not being spawned

### PR DESCRIPTION
**Pull request by kyujin-cho**
Successor of #1251.

### CODE REVIEW

**Review by inureyes — APPROVED** [review]
LGTM

File: src/ai/backend/manager/models/scaling_group.py [hunk 1]
```diff
@@ -377,7 +383,9 @@ async def load_by_group(
             scaling_groups.c.name == sgroups_for_groups.c.scaling_group,
         )
         query = (
-            sa.select([scaling_groups]).select_from(j).where(sgroups_for_groups.c.group == group)
+            sa.select([scaling_groups])
+            .select_from(j)
+            .where(sgroups_for_groups.c.group == group and scaling_groups.c.is_public.is_(True))
```
Comment (agatha197) [inline src/ai/backend/manager/models/scaling_group.py]: This code can raise TypeError("Boolean value of this clause is not defined")
```suggestion
            .where(sgroups_for_groups.c.group & group and scaling_groups.c.is_public.is_(True))
```
ref: https://stackoverflow.com/questions/42681231/sqlalchemy-unexpected-results-when-using-and-and-or

File: src/ai/backend/manager/models/scaling_group.py [hunk 2]
```diff
@@ -352,7 +356,9 @@ async def load_by_domain(
             scaling_groups.c.name == sgroups_for_domains.c.scaling_group,
         )
         query = (
-            sa.select([scaling_groups]).select_from(j).where(sgroups_for_domains.c.domain == domain)
+            sa.select([scaling_groups])
+            .select_from(j)
+            .where(sgroups_for_domains.c.domain == domain and scaling_groups.c.is_public.is_(True))
```
Comment (agatha197) [inline src/ai/backend/manager/models/scaling_group.py]: same as below

File: src/ai/backend/manager/models/scaling_group.py [hunk 3]
```diff
@@ -404,7 +412,10 @@ async def load_by_keypair(
         query = (
             sa.select([scaling_groups])
             .select_from(j)
-            .where(sgroups_for_keypairs.c.access_key == access_key)
+            .where(
+                sgroups_for_keypairs.c.access_key == access_key
+                and scaling_groups.c.is_public.is_(True)
```
Comment (agatha197) [inline src/ai/backend/manager/models/scaling_group.py]: Same as above.

File: src/ai/backend/manager/models/scaling_group.py [hunk 4]
```diff
@@ -429,7 +440,9 @@ async def batch_load_by_group(
         query = (
             sa.select([scaling_groups, sgroups_for_groups.c.group])
             .select_from(j)
-            .where(sgroups_for_groups.c.group.in_(group_ids))
+            .where(
+                sgroups_for_groups.c.group.in_(group_ids) and scaling_groups.c.is_public.is_(True)
```
Comment (agatha197) [inline src/ai/backend/manager/models/scaling_group.py]: Same as above

File: src/ai/backend/manager/models/scaling_group.py [hunk 5]
```diff
@@ -450,7 +463,7 @@ async def batch_load_by_name(
         query = (
             sa.select([scaling_groups])
             .select_from(scaling_groups)
-            .where(scaling_groups.c.name.in_(names))
+            .where(scaling_groups.c.name.in_(names) and scaling_groups.c.is_public.is_(True))
```
Comment (agatha197) [inline src/ai/backend/manager/models/scaling_group.py]: Same as above

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]
I found TypeError.

**Pull request merged**